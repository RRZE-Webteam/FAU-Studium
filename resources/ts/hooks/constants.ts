export const FACULTY_PHILOSOPHY = 'Phil';
export const FACULTY_NATURAL_SCIENCES = 'Nat';
export const SEMESTER_SUMMER = 'Sommersemester';
export const ADMISSION_REQUIREMENT_FREE = 'frei';
