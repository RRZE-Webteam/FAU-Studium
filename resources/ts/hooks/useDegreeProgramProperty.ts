import produce, { Immutable } from 'immer';
import { get, set } from 'lodash';

import { useEntityProp } from '@wordpress/core-data';

import serverData from 'util/serverData';

import { DegreeProgramDataPaths } from '../defs';

/**
 * Hook that returns the value and a setter for the
 * specified property of the current DegreeProgramData.
 * Should be used only for single DegreeProgram edit screen.
 *
 * To make immutable changes easy, Immer library is used.
 *
 * @param path Path to DegreeProgramData value.
 * @typeParam Value Definition of returned DegreeProgramData property.
 * @returns An array where the first item is the property value, and the second is the setter.
 *
 * @example
 * ```
 * function MultilingualInput() {
 *     const [englishTitle, setEnglishTitle] = useDegreeProgramProperty<MultilingualString>('title.en');
 *     return <Input description="English title"
 *            value={englishTitle}
 *            onClick={setEnglishTitle} />
 * }
 * ```
 */
export default function useDegreeProgramProperty<Value>(
    path: DegreeProgramDataPaths,
): [Immutable<undefined | Value>, (value: Value) => void] {
    const [degreeProgramData, setDegreeProgramData] = useEntityProp(
        'postType',
        serverData().postType,
        serverData().propertyName,
    );

    return [
        get(degreeProgramData, path),
        (val: Value) =>
            setDegreeProgramData(
                produce<Value>(degreeProgramData, (draft) => set(draft, path, val)),
            ),
    ];
}
