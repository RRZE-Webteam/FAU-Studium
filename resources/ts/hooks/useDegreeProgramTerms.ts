import { store } from '@wordpress/core-data';
import { useSelect } from '@wordpress/data';

import serverData from 'util/serverData';

import {
    CoreDataSelectors,
    DegreeMeta,
    DegreeProgramEditorServerDataType,
    EnglishNameMeta,
    MapSelect,
    MultilingualLinkMeta,
    WpTerm,
} from '../defs';

type TaxonomySlug = keyof DegreeProgramEditorServerDataType['taxonomySlugs'];
type Term<T> = T extends
    | 'areaOfStudy'
    | 'bachelorOrTeachingDegreeAdmissionRequirement'
    | 'examinationsOffice'
    | 'faculty'
    | 'germanLanguageSkillsForInternationalStudents'
    | 'masterDegreeAdmissionRequirement'
    | 'subjectSpecificAdvice'
    | 'teachingDegreeHigherSemesterAdmissionRequirement'
    ? WpTerm<MultilingualLinkMeta>
    : T extends
          | 'attribute'
          | 'keyword'
          | 'semester'
          | 'studyLocation'
          | 'subjectGroup'
          | 'teachingLanguage'
    ? WpTerm<EnglishNameMeta>
    : T extends 'degree'
    ? WpTerm<DegreeMeta>
    : T extends 'numberOfStudents'
    ? WpTerm
    : never;

/**
 * Hook that returns Degree Program terms for selected taxonomy.
 *
 * @param taxonomy Taxonomy name. One of DegreeProgramEditorServerDataType['taxonomySlugs'] keys.
 * @returns Terms array.
 *
 * @example
 * ```
 * function MultilingualSelect() {
 *     const areaOfStudyTermsList = useDegreeProgramTerms('areaOfStudy');
 *     areaOfStudyTermsList.forEach((term) => {
 *         options.push({
 *             value: term.id,
 *             label: term.meta.name_en + term.meta.link_text_en
 *         })
 *     })
 *
 *     return <SelectControl options={options} />;
 * }
 * ```
 */
export default function useDegreeProgramTerms<T extends TaxonomySlug>(taxonomy: T): Array<Term<T>> {
    return useSelect<MapSelect<CoreDataSelectors>>(
        (select) => {
            const { getEntityRecords } = select(store.name);
            return getEntityRecords('taxonomy', serverData().taxonomySlugs[taxonomy]) || [];
        },
        [taxonomy],
    );
}
