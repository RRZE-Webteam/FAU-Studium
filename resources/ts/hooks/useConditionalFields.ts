import {
    ADMISSION_REQUIREMENT_FREE,
    FACULTY_NATURAL_SCIENCES,
    FACULTY_PHILOSOPHY,
    SEMESTER_SUMMER,
} from './constants';
import useDegreeProgramProperty from './useDegreeProgramProperty';

import { Degree, DEGREE_ABBREVIATION_GERMAN, MultilingualLink } from 'defs';

const ALLOWED_FACULTIES_FOR_COMBINATION = [FACULTY_PHILOSOPHY, FACULTY_NATURAL_SCIENCES];

export function useDegreeFeesEnabled() {
    const [feeRequired] = useDegreeProgramProperty<boolean>('fee_required');

    return !!feeRequired;
}

export function useCombinationOfDegreeProgramEnabled() {
    const [faculty] = useDegreeProgramProperty<MultilingualLink>('faculty');
    const [degree] = useDegreeProgramProperty<Degree>('degree');

    if (!faculty || !degree) {
        return false;
    }

    return (
        ALLOWED_FACULTIES_FOR_COMBINATION.includes(faculty.name.de) &&
        degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.BACHELOR
    );
}

export function useAdmissionRequirementsForBachelorAndTeachingDegreesEnable() {
    const [degree] = useDegreeProgramProperty<Degree>('degree');

    if (!degree) {
        return false;
    }

    return (
        degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.BACHELOR ||
        degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.TEACHING_DEGREE
    );
}

export function useAdmissionRequirementsTeachingDegreeAtHigherSemesterEnabled() {
    const [bachelorOrTeachingAdmissionRequirement] = useDegreeProgramProperty<MultilingualLink>(
        'admission_requirements.bachelor_or_teaching_degree',
    );

    if (!bachelorOrTeachingAdmissionRequirement) {
        return false;
    }

    return bachelorOrTeachingAdmissionRequirement.name.de !== ADMISSION_REQUIREMENT_FREE;
}

export function useAdmissionRequirementsForMastersDegree() {
    const [degree] = useDegreeProgramProperty<Degree>('degree');

    if (!degree) {
        return false;
    }

    return degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.MASTERS;
}

export function useLanguageSkillsForFacultyOfHumanitiesOnlyEnabled() {
    const [degree] = useDegreeProgramProperty<Degree>('degree');
    const [faculty] = useDegreeProgramProperty<MultilingualLink>('faculty');

    if (!degree || !faculty) {
        return false;
    }

    return (
        (degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.BACHELOR ||
            degree.abbreviation.de === DEGREE_ABBREVIATION_GERMAN.TEACHING_DEGREE) &&
        faculty.name.de === FACULTY_PHILOSOPHY
    );
}

export function useApplicationDeadlineSummerSemesterEnabled() {
    const [semester] = useDegreeProgramProperty<Degree>('semester_dates');

    if (!semester) {
        return false;
    }

    return semester.name.de === SEMESTER_SUMMER;
}
