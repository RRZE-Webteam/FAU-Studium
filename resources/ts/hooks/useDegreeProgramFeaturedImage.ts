import { useEffect } from 'react';
import { Image } from 'defs';
import serverData from 'util/serverData';

import { useEntityProp } from '@wordpress/core-data';
import { useSelect } from '@wordpress/data';

import useDegreeProgramProperty from './useDegreeProgramProperty';

export default function useDegreeProgramFeaturedImage(): [Image, (val: Image) => void] {
    const [postThumbnail] = useEntityProp('postType', serverData.postType, 'featured_media');
    const [featuredImage, setFeaturedImage] = useDegreeProgramProperty<Image>('featured_image');

    if (!featuredImage) {
        return [
            {
                id: 0,
                url: '',
            },
            setFeaturedImage,
        ];
    }

    const thumbnailMedia = useSelect(
        (select: any) => {
            return select('core').getMedia(postThumbnail);
        },
        [postThumbnail],
    );

    useEffect(() => {
        setFeaturedImage({
            id: postThumbnail,
            url: thumbnailMedia?.media_details?.sizes?.thumbnail?.source_url ?? '',
        });
    }, [postThumbnail, thumbnailMedia]);

    return [featuredImage, setFeaturedImage];
}
