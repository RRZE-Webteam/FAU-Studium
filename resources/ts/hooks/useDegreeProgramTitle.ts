import { Immutable } from 'immer';

import { useEntityProp } from '@wordpress/core-data';

import ServerData from '../util/serverData';
import useDegreeProgramProperty from './useDegreeProgramProperty';

type GermanTitle = string;
type EnglishTitle = string;
export default function useDegreeProgramTitle(): {
    germanTitle: Immutable<GermanTitle | undefined>;
    setGermanTitle: (title: GermanTitle) => void;
    englishTitle: Immutable<EnglishTitle | undefined>;
    setEnglishTitle: (title: EnglishTitle) => void;
} {
    const [, setDegreeProgramTitleGerman] = useDegreeProgramProperty<string>('title.de');
    const [degreeProgramTitleEnglish, setDegreeProgramTitleEnglish] =
        useDegreeProgramProperty<string>('title.en');
    const [postTitle, setPostTitle] = useEntityProp('postType', ServerData.postType, 'title');

    return {
        germanTitle: postTitle,
        setGermanTitle: (title: GermanTitle) => {
            setPostTitle(title);
            setDegreeProgramTitleGerman(title);
        },
        englishTitle: degreeProgramTitleEnglish,
        setEnglishTitle: (title: EnglishTitle) => {
            setDegreeProgramTitleEnglish(title);
        },
    };
}
