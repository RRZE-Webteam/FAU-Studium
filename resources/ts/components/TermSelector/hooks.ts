import { useMemo } from 'react';
import { TaxonomySlug } from 'defs/term';
import useDegreeProgramTerms from 'hooks/useDegreeProgramTerms';

import {} from '@wordpress/components/src/combobox-control';

export interface SelectOption {
    label: string;
    value: string;
}

export function useTermsListAsSelectOptions(taxonomy: TaxonomySlug): SelectOption[] {
    const terms = useDegreeProgramTerms(taxonomy);

    const options = useMemo(
        () =>
            terms.map((term) => ({
                label: term.name,
                value: term.id.toString(),
            })),
        [terms],
    );

    return options;
}

export default {};
