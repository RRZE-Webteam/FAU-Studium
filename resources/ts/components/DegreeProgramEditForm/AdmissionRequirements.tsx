import React from 'react';
import styled from 'styled-components';

import { __experimentalHeading as Heading, BaseControl } from '@wordpress/components';
import { _x } from '@wordpress/i18n';

import FormFieldWrapper from 'components/Layouts/FormFieldWrapper';
import FormWrapper from 'components/Layouts/FormWrapper';
import TermSelector from 'components/TermSelector';
import { useEditDegreeProgram } from 'contexts/DegreeProgramEditFormProvider';
import {
    useAdmissionRequirementsForBachelorAndTeachingDegreesEnable,
    useAdmissionRequirementsForMastersDegree,
    useAdmissionRequirementsTeachingDegreeAtHigherSemesterEnabled,
    useApplicationDeadlineSummerSemesterEnabled,
    useLanguageSkillsForFacultyOfHumanitiesOnlyEnabled,
} from 'hooks/useConditionalFields';
import { transformTermToMultilingualLink } from 'util/transforms';

import ContentField from '../ContentField';
import MultilingualContainer from './MultilingualContainer';

import { MultilingualLink } from 'defs';

const StyledHeading = styled(Heading)`
    margin-bottom: 12px;
`;

const AdmissionRequirements = () => {
    const { values, handleChange } = useEditDegreeProgram();
    const bachelorOrTeachingEnabled = useAdmissionRequirementsForBachelorAndTeachingDegreesEnable();
    const teachingDegreeHigherSemesterEnabled =
        useAdmissionRequirementsTeachingDegreeAtHigherSemesterEnabled();
    const mastersEnabled = useAdmissionRequirementsForMastersDegree();
    const humanitiesFacultyLanguageSkillsEnabled =
        useLanguageSkillsForFacultyOfHumanitiesOnlyEnabled();
    const applicationSummerDeadlineEnabled = useApplicationDeadlineSummerSemesterEnabled();

    return (
        <>
            <div>
                <StyledHeading level={4}>Bachelor/Lehramt</StyledHeading>

                <FormWrapper>
                    {bachelorOrTeachingEnabled && (
                        <FormFieldWrapper fill="half">
                            <TermSelector
                                id="bachelor_teaching_admission_requirement"
                                label={_x(
                                    "Admission requirements for Bachelor's/teaching degrees",
                                    'backoffice: degree program edit form',
                                    'fau-degree-program',
                                )}
                                taxonomy="bachelorOrTeachingDegreeAdmissionRequirement"
                                value={values.admission_requirements.bachelor_or_teaching_degree.id}
                                onChange={(term) => {
                                    handleChange<MultilingualLink>(
                                        'admission_requirements.bachelor_or_teaching_degree',
                                        transformTermToMultilingualLink(term),
                                    );
                                }}
                            />
                        </FormFieldWrapper>
                    )}

                    {teachingDegreeHigherSemesterEnabled && (
                        <FormFieldWrapper fill="half">
                            <TermSelector
                                id="teaching_higher_semester_admission_requirement"
                                label={_x(
                                    'Admission requirements for entering a teaching degree at a higher semester',
                                    'backoffice: degree program edit form',
                                    'fau-degree-program',
                                )}
                                taxonomy="teachingDegreeHigherSemesterAdmissionRequirement"
                                value={
                                    values.admission_requirements.teaching_degree_higher_semester.id
                                }
                                onChange={(term) => {
                                    handleChange<MultilingualLink>(
                                        'admission_requirements.teaching_degree_higher_semester',
                                        transformTermToMultilingualLink(term),
                                    );
                                }}
                            />
                        </FormFieldWrapper>
                    )}
                </FormWrapper>
            </div>

            <div>
                <StyledHeading level={4}>Master</StyledHeading>

                {mastersEnabled && (
                    <TermSelector
                        id="master_admission_requirement"
                        label={_x(
                            "Admission requirements for Master's degree",
                            'backoffice: degree program edit form',
                            'fau-degree-program',
                        )}
                        taxonomy="masterDegreeAdmissionRequirement"
                        value={values.admission_requirements.master.id}
                        onChange={(term) => {
                            handleChange<MultilingualLink>(
                                'admission_requirements.master',
                                transformTermToMultilingualLink(term),
                            );
                        }}
                    />
                )}

                <BaseControl
                    label={_x(
                        "Content-related admission requirements for Master's degree",
                        'backoffice: degree program edit form',
                        'fau-degree-program',
                    )}
                    help="vorausgegangene Bachelorstudieng채nge, Qualifikationswerte, etc."
                >
                    <MultilingualContainer value={values.content_related_master_requirements}>
                        {(languageCode) => (
                            <ContentField
                                onChange={(value: string) => {
                                    handleChange<string>(
                                        `content_related_master_requirements.${languageCode}`,
                                        value,
                                    );
                                }}
                                content={values.content_related_master_requirements[languageCode]}
                            />
                        )}
                    </MultilingualContainer>
                </BaseControl>

                <BaseControl
                    label={_x(
                        'Application deadline winter semester',
                        'backoffice: degree program edit form',
                        'fau-degree-program',
                    )}
                >
                    <ContentField
                        onChange={(value: string) => {
                            handleChange<string>('application_deadline_winter_semester', value);
                        }}
                        content={values.application_deadline_winter_semester}
                    />
                </BaseControl>

                {applicationSummerDeadlineEnabled && (
                    <BaseControl
                        label={_x(
                            'Application deadline summer semester',
                            'backoffice: degree program edit form',
                            'fau-degree-program',
                        )}
                    >
                        <ContentField
                            onChange={(value: string) => {
                                handleChange<string>('application_deadline_summer_semester', value);
                            }}
                            content={values.application_deadline_summer_semester}
                        />
                    </BaseControl>
                )}
            </div>

            <div>
                <StyledHeading level={4}>F체r Bachelor/Lehramt/Master</StyledHeading>

                <BaseControl
                    label={_x(
                        'Details and notes',
                        'backoffice: degree program edit form',
                        'fau-degree-program',
                    )}
                    help="evtl. zus채tzliche Infos zu Bewerbung und Einschreibung."
                >
                    <MultilingualContainer value={values.details_and_notes}>
                        {(languageCode) => (
                            <ContentField
                                onChange={(value: string) => {
                                    handleChange<string>(
                                        `details_and_notes.${languageCode}`,
                                        value,
                                    );
                                }}
                                content={values.details_and_notes[languageCode]}
                            />
                        )}
                    </MultilingualContainer>
                </BaseControl>

                <BaseControl
                    label={_x(
                        'Language skills',
                        'backoffice: degree program edit form',
                        'fau-degree-program',
                    )}
                    help="Sprachkenntnisse, die f체r den Studiengang erforderlich sind."
                >
                    <MultilingualContainer value={values.language_skills}>
                        {(languageCode) => (
                            <ContentField
                                onChange={(value: string) => {
                                    handleChange<string>(`language_skills.${languageCode}`, value);
                                }}
                                content={values.language_skills[languageCode]}
                            />
                        )}
                    </MultilingualContainer>
                </BaseControl>

                {humanitiesFacultyLanguageSkillsEnabled && (
                    <BaseControl
                        label={_x(
                            'Language skills for Faculty of Humanities, Social Sciences, and Theology only',
                            'backoffice: degree program edit form',
                            'fau-degree-program',
                        )}
                        help="Sprachkenntnisse, die auch in den ersten Semestern noch erworben werden k철nnen."
                    >
                        <ContentField
                            onChange={(value: string) => {
                                handleChange<string>('language_skills_humanities_faculty', value);
                            }}
                            content={values.language_skills_humanities_faculty}
                        />
                    </BaseControl>
                )}

                <TermSelector
                    id="german_language_skills_for_international_students"
                    label={_x(
                        'Language certificates/German language skills for international applicants',
                        'backoffice: degree program edit form',
                        'fau-degree-program',
                    )}
                    taxonomy="germanLanguageSkillsForInternationalStudents"
                    value={values.german_language_skills_for_international_students.id}
                    onChange={(term) => {
                        handleChange<MultilingualLink>(
                            'german_language_skills_for_international_students',
                            transformTermToMultilingualLink(term),
                        );
                    }}
                />
            </div>
        </>
    );
};

export default AdmissionRequirements;
