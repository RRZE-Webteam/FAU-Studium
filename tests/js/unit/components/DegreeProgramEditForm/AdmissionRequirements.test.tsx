import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render, screen } from '@testing-library/react';

import AdmissionRequirements from 'components/DegreeProgramEditForm/AdmissionRequirements';
import * as useConditionalFields from 'hooks/useConditionalFields';

jest.mock('nanoid', () => ({ nanoid: () => `key_${Math.random()}` }));
jest.mock('hooks/useConditionalFields', () => {
    const conditionalHooks = jest.requireActual('hooks/useConditionalFields');
    return Object.keys(conditionalHooks).reduce(
        (object, key) => ({ ...object, [key]: jest.fn(() => false) }),
        {},
    );
});

jest.mock(
    'components/TermSelector',
    // eslint-disable-next-line react/prop-types
    jest.fn(() =>
        jest.fn(({ label }) => (
            <form>
                <label htmlFor="foo">
                    <span>{label}</span>
                    <input aria-labelledby="foo" />
                </label>
            </form>
        )),
    ),
);
jest.mock('components/DegreeProgramEditForm/MultilingualContainer', () => () => <></>);
jest.mock('components/ContentField', () => () => <></>);
jest.mock('contexts/DegreeProgramEditFormProvider', () => ({
    useEditDegreeProgram: () => ({
        values: {
            admission_requirements: {
                bachelor_or_teaching_degree: {
                    id: '',
                },
                teaching_degree_higher_semester: {
                    id: '',
                },
                master: {
                    id: '',
                },
            },
            content_related_master_requirements: {
                id: '',
                de: '',
                en: '',
            },
            language_skills: {
                id: '',
                de: '',
                en: '',
            },
            details_and_notes: {
                id: '',
                de: '',
                en: '',
            },
            german_language_skills_for_international_students: {
                id: '',
            },
            degree: {
                id: '',
                name: {
                    id: '',
                    de: '',
                    en: '',
                },
                abbreviation: {
                    id: '',
                    de: '',
                    en: '',
                },
            },
            language_skills_humanities_faculty: '',
            application_deadline_winter_semester: '',
        },
        handleChange: () => {},
    }),
}));

jest.mock('contexts/DegreeProgramValidationProvider', () => ({
    useValidation: () => ({
        errors: new Map(),
        getFieldErrors: jest.fn(() => new Map()),
        removeError: jest.fn(() => {}),
        fieldHasErrors: jest.fn(() => {}),
    }),
}));

describe('Admission requirement fields', () => {
    afterEach(() => {
        cleanup();
    });

    it('should not include conditional fields by default', () => {
        render(<AdmissionRequirements />);

        expect(screen.queryByText(/Bachelor\/Lehramt[^/]/i)).toBeNull();
        expect(
            screen.queryByText(/Admission requirements for Bachelor's\/teaching degrees/i),
        ).toBeNull();
        expect(
            screen.queryByText(
                /Admission requirements for entering a Bachelor's\/teaching degree at a higher semester/i,
            ),
        ).toBeNull();
        expect(screen.queryByText(/^Admission requirements for Master's degree/i)).toBeNull();
        expect(screen.queryByText(/Application deadline summer semester/i)).toBeNull();
        expect(
            screen.queryByText(
                /Language skills for Faculty of Humanities, Social Sciences, and Theology only/i,
            ),
        ).toBeNull();
    });

    it('should include conditional fields when conditions met', () => {
        Object.keys(useConditionalFields).forEach((hookName) => {
            if (hookName === 'default') {
                return;
            }

            jest.spyOn(
                useConditionalFields,
                hookName as keyof typeof useConditionalFields,
            ).mockReturnValue(true);
        });

        render(<AdmissionRequirements />);

        expect(
            screen.queryByText(/Admission requirements for Bachelor's\/teaching degrees/i),
        ).toBeInTheDocument();
        expect(
            screen.queryByText(
                /Admission requirements for entering a Bachelor's\/teaching degree at a higher semester/i,
            ),
        ).toBeInTheDocument();
        expect(
            screen.queryByText(/^Admission requirements for Master's degree/i),
        ).toBeInTheDocument();
        expect(screen.queryByText(/Application deadline summer semester/i)).toBeInTheDocument();
        expect(
            screen.queryByText(
                /Language skills for Faculty of Humanities, Social Sciences, and Theology only/i,
            ),
        ).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const test = renderer.create(<AdmissionRequirements />).toJSON();

        expect(test).toMatchSnapshot();
    });
});
