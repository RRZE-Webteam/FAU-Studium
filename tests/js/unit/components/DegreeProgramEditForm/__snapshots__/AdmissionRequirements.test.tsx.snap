// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admission requirement fields should match snapshot 1`] = `
<div
  className="components-panel"
>
  <div
    className="components-panel__body is-opened"
  >
    <div
      className="components-flex sc-hLBbgP bCBYrY css-zp26ix-View-Flex-base-ItemsRow e19lxcc00"
      data-wp-c16t={true}
      data-wp-component="Flex"
    >
      <div
        className="components-flex-item sc-dkrFOg dXOgOj css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="half"
      >
        <form>
          <label
            htmlFor="foo"
          >
            <span>
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Admission requirements for Bachelor's/teaching degrees 
              </span>
            </span>
            <input
              aria-labelledby="foo"
            />
          </label>
        </form>
      </div>
      <div
        className="components-flex-item sc-dkrFOg dXOgOj css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="half"
      >
        <form>
          <label
            htmlFor="foo"
          >
            <span>
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Admission requirements for entering a teaching degree at a higher semester 
              </span>
            </span>
            <input
              aria-labelledby="foo"
            />
          </label>
        </form>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <hr />
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <form>
          <label
            htmlFor="foo"
          >
            <span>
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Admission requirements for Master's degree
              </span>
            </span>
            <input
              aria-labelledby="foo"
            />
          </label>
        </form>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Content-related admission requirements for Master's degree
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                vorausgegangene Bachelorstudiengänge, Qualifikationswerte, etc.
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <hr />
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Application deadline winter semester
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                Datum im Format TT.MM. angeben. Weitere Termine können im Feld „Details und Anmerkungen“ eingetragen werden.
              </p>
            </span>
            <div
              className="components-base-control c-day-month-picker css-ij2po-Wrapper-boxSizingReset ej5x27r4"
            >
              <div
                className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
              >
                <input
                  className="components-text-control__input"
                  id="inspector-text-control-2"
                  onChange={[Function]}
                  pattern="((0[1-9]|[12][0-9]|3[01])\\.(0[13578]|1[02])|(0[1-9]|[12][0-9]|30)\\.(0[469]|11)|(0[1-9]|1[0-9]|2[0-8])\\.02)\\."
                  placeholder="TT.MM."
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Application deadline summer semester
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                Datum im Format TT.MM. angeben. Weitere Termine können im Feld „Details und Anmerkungen“ eingetragen werden.
              </p>
            </span>
            <div
              className="components-base-control c-day-month-picker css-ij2po-Wrapper-boxSizingReset ej5x27r4"
            >
              <div
                className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
              >
                <input
                  className="components-text-control__input"
                  id="inspector-text-control-3"
                  onChange={[Function]}
                  pattern="((0[1-9]|[12][0-9]|3[01])\\.(0[13578]|1[02])|(0[1-9]|[12][0-9]|30)\\.(0[469]|11)|(0[1-9]|1[0-9]|2[0-8])\\.02)\\."
                  placeholder="TT.MM."
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <hr />
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Details and notes
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                evtl. zusätzliche Infos zu Bewerbung und Einschreibung.
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Language skills
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                Sprachkenntnisse, die für den Studiengang erforderlich sind.
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <div
          className="components-base-control css-ij2po-Wrapper-boxSizingReset ej5x27r4"
        >
          <div
            className="components-base-control__field css-10urnh1-StyledField-deprecatedMarginField ej5x27r3"
          >
            <span
              className="components-base-control__label css-1bhsfm8-StyledVisualLabel-baseLabelTypography-labelStyles ej5x27r0"
            >
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Language skills for Faculty of Humanities, Social Sciences, and Theology only 
              </span>
              <p
                className="sc-gswNZR hgMeoo"
              >
                Sprachkenntnisse, die auch in den ersten Semestern noch erworben werden können.
              </p>
            </span>
          </div>
        </div>
      </div>
      <div
        className="components-flex-item sc-dkrFOg kpsgMX css-mw3lhz-View-Item-sx-Base e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="FlexItem"
        fill="full"
      >
        <form>
          <label
            htmlFor="foo"
          >
            <span>
              <span
                className="sc-bcXHqe bKtfDB"
              >
                Language certificates/German language skills for international applicants
              </span>
            </span>
            <input
              aria-labelledby="foo"
            />
          </label>
        </form>
      </div>
    </div>
  </div>
</div>
`;
